src/test/main.ml

```diff
@@ -763,6 +756,44 @@ let tests =
           return 3
         ]
       );
+    exits 2 ~name:"redirect-fails" Genspio.EDSL.(
+        let tmp1 = tmp_file "fd3" in
+        let tmp2 = tmp_file "return" in
+        let recognizable = "heelllloooooo" in
+        let this_is_bash =
+          (exec ["ps"] |> output_as_string)
+          >> exec ["grep"; "bash"]
+          |> returns ~value:0 in
+        seq [
+          tmp1#set (string "");
+          write_output
+            ~return_value:tmp2#path (
+            with_redirections (exec ["printf"; "%s"; recognizable]) [
+              to_fd (int 4) (int 3);
```

Comment:
This redirection is the one that fails, since fd `3` is not yet opened.

In the POSIX case, `with_redirections` then exits, in the bash/zsh case it keeps going.